src/writer/workflows_blocks/salesforceintegration.py

```diff
@@ -0,0 +1,330 @@
+            self.outcome = "error"
+            raise RuntimeError(f"Update error: {str(e)}")
+
+    def _delete_record(self, object_name: str, record_id: str) -> Dict[str, Any]:
+        """Delete a record"""
+        try:
+            obj = getattr(self.sf, object_name)
+            result = obj.delete(record_id)
+            return result
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Delete error: {str(e)}")
+
+    def _upsert_record(self, object_name: str, external_id_field: str, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Upsert a record using external ID"""
+        try:
+            obj = getattr(self.sf, object_name)
+            result = obj.upsert(external_id_field, data)
+            return result
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Upsert error: {str(e)}")
+
+    def _bulk_operation(self, operation: str, object_name: str, data: List[Dict[str, Any]], 
+                       batch_size: int = 200) -> Dict[str, Any]:
+        """Execute bulk operation"""
+        try:
+            results = []
+            for i in range(0, len(data), batch_size):
+                batch = data[i:i + batch_size]
+                
+                if operation == "create":
+                    result = getattr(self.sf.bulk, object_name).insert(batch)
+                elif operation == "update":
+                    result = getattr(self.sf.bulk, object_name).update(batch)
+                elif operation == "delete":
+                    result = getattr(self.sf.bulk, object_name).delete(batch)
+                elif operation == "upsert":
+                    result = getattr(self.sf.bulk, object_name).upsert(batch)
+                
+                results.extend(result)
+            
+            return {
+                "total_processed": len(results),
+                "results": results
+            }
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Bulk operation error: {str(e)}")
+
+    def run(self):
+        try:
+            # Get authentication fields
+            username = self._get_field("username")
+            password = self._get_field("password")
+            security_token = self._get_field("security_token")
+            domain = self._get_field("domain", False, "login")
+
+            # Connect to Salesforce
+            self._connect(username, password, security_token, domain)
+
+            # Get operation details
+            operation = self._get_field("operation")
+            object_name = self._get_field("object_name", True)
+            record_id = self._get_field("record_id", True)
+            data = self._get_field("data", True, "{}")
+            query = self._get_field("query", True)
+            external_id_field = self._get_field("external_id_field", True)
+            batch_size = int(self._get_field("batch_size", True, "200"))
```

Comment:
Some fields are not JSON
```suggestion
            object_name = self._get_field("object_name")
            record_id = self._get_field("record_id")
            data = self._get_field("data", True, "{}")
            query = self._get_field("query")
            external_id_field = self._get_field("external_id_field")
            batch_size = int(self._get_field("batch_size", False, "200"))
```